inference-engine/src/plaidml_plugin/ops/variadic_split.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset1.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+
+namespace PlaidMLPlugin {
+
+static OpRegistration reg("variadicsplit", [](const Context& ctx) {
+  IE_ASSERT(ctx.operands.size() == 3);
+  auto I = ctx.operands.at(0);
+  auto axes = get_axis_vector_from_constant_operand(1, ctx.layer);
+  IE_ASSERT(axes.size() == 1);
+  auto axis = axes[0];
+  auto split_lengths = get_shape_from_constant_operand(2, ctx.layer);
+
+  auto ndims = I.rank();
+  std::vector<edsl::TensorDim> I_dims(ndims);
+  std::vector<edsl::TensorIndex> I_idxs(ndims);
+  std::vector<edsl::Tensor> Os;
+  I.bind_dims(I_dims);
+  auto O_dims = I_dims;
+
+  size_t split_size = 0;
+  for (auto split : split_lengths) {
+      if (split != -1) {
+        split_size += split;
+      }
+  }
+  auto placeholder = I_dims[axis] - split_size;
+
+  edsl::TensorDim offset(0);
+  for (auto split : split_lengths) {
+    auto O_idxs = I_idxs;
+    O_idxs[axis] = I_idxs[axis] - offset;
+    if (split == -1) {
+      O_dims[axis] = placeholder;
+      offset = offset + placeholder;
+    } else {
+      O_dims[axis] = static_cast<edsl::TensorDim>(split);
```

Comment:
A style nitpick: Elsewhere when we want to make a `TensorDim` from an integer, we construct a `TensorDim` (i.e. with `edsl::TensorDim(some_integer)`) rather than `static_cast`ing the integer to a `TensorDim`

... although it now occurs to me that that style decision may not have been deliberate, so we could instead change to `static_cast`ing elsewhere (there aren't that many cases)